rfcs/proposed/task_group_dynamic_dependencies/README.md

```diff
@@ -366,13 +340,26 @@ This task tree matches the one shown earlier for merge-sort.
 
 ## Open Questions in Design
 
-Some open questions that remain:
-
+Questions that should be addressed before moving into `experimental` and providing a preview feature:
 - Are the suggested APIs sufficient?
+- Are there additional use cases that should be considered that we missed in our analysis?
+- Semantics for `task_arena::enqueue` method that takes a `task_handle` object should be defined for all possible states of the handle.
+- What are the performance targets for this feature?
+- What are the exit criteria for a feature to become fully supported?
+- The API applicability for other usage scenarios should be checked, e.g.:
+  * A wavefront example representing tasks with multiple successors and predecessors
+  * Fibonacci example
+  * A two-stage parallel scan
+  * Other non-trivial divide-and-conquer patterns
+
+Questions that should be addressed before the future movement into `supported` and having a fully supported feature:
 - Should we add a function to adds more than one predecessor as single call, such as `add_predecessors`?
 - Should we add functions that merge creation and definition of predecessor tasks, such as
-`template <typename Func> add_predecessor(Func&& f);`.
-- Are there additional use cases that should be considered that we missed in our analysis?
+`template <typename Func> add_predecessor(Func&& f);`?
+- Should we add a function that indicates that the task handled by `task_handle` object is completed, such as
+`bool task_handle::is_completed()`?
```

Comment:
Is "completed" the only state which we think can be of interest? For example, might be "submitted'" is also good to know.

Reply:
I think it may be useful, but only if we consider task in the following states:
* Not submitted and hence not completed (just created)
* Submitted and maybe completed (`run` or `run_and_wait` was called)
* Submitted and completed

If we consider the state to always change "created -> submitted -> completed", having the checker for submitted state seems useless since the state can be immediately changed from submitted to completed while returning the state from the function. 

I have added this to the document for further consideration.